tests/golos.vesting_tests.cpp

```diff
@@ -413,6 +419,264 @@ BOOST_FIXTURE_TEST_CASE( test_cancel_convert_vesting_to_token, golos_vesting_tes
                              ("balance", "405.0000 GOLOS") );
 } FC_LOG_AND_RETHROW()
 
+BOOST_FIXTURE_TEST_CASE( error_sender_equals_test_delegate_vesting, golos_vesting_tester ) try {
+    BOOST_REQUIRE_EQUAL( success(), create(N(golos.issuer), asset::from_string("100000.0000 GOLOS")) );
+    BOOST_REQUIRE_EQUAL( success(), issue(N(golos.issuer), N(sania), asset::from_string("500.0000 GOLOS"), "issue tokens sania") );
+    BOOST_REQUIRE_EQUAL( success(), issue(N(golos.issuer), N(pasha), asset::from_string("500.0000 GOLOS"), "issue tokens pasha") );
+    produce_blocks(1);
+
+    BOOST_REQUIRE_EQUAL( success(), open_balance(N(sania), symbol(4,"GOLOS"), N(sania)) );
+    BOOST_REQUIRE_EQUAL( success(), open_balance(N(pasha), symbol(4,"GOLOS"), N(pasha)) );
+    produce_blocks(1);
+
+    BOOST_REQUIRE_EQUAL( success(), create_vesting_token(N(golos.issuer), symbol(4,"GOLOS")) );
+    BOOST_REQUIRE_EQUAL( success(), transfer(N(sania), N(golos.vest), asset::from_string("100.0000 GOLOS"), "convert token to vesting") );
+    BOOST_REQUIRE_EQUAL( success(), transfer(N(pasha), N(golos.vest), asset::from_string("100.0000 GOLOS"), "convert token to vesting") );
+    produce_blocks(1);
+
+    auto sania_token_balance = get_account(N(sania), "4,GOLOS");
+    REQUIRE_MATCHING_OBJECT( sania_token_balance, mvo()
+                             ("balance", "400.0000 GOLOS") );
+
+    auto pasha_token_balance = get_account(N(pasha), "4,GOLOS");
+    REQUIRE_MATCHING_OBJECT( pasha_token_balance, mvo()
+                             ("balance", "400.0000 GOLOS") );
+
+    auto sania_vesting_balance = get_account_vesting(N(sania), "4,GOLOS");
+    REQUIRE_MATCHING_OBJECT( sania_vesting_balance, mvo()
+                             ("vesting", "100.0000 GOLOS")
+                             ("delegate_vesting", "0.0000 GOLOS")
+                             ("received_vesting", "0.0000 GOLOS")
+                             ("unlocked_limit", "0.0000 GOLOS")
+                             );
+
+    auto pasha_vesting_balance = get_account_vesting(N(pasha), "4,GOLOS");
+    REQUIRE_MATCHING_OBJECT( pasha_vesting_balance, mvo()
+                             ("vesting", "100.0000 GOLOS")
+                             ("delegate_vesting", "0.0000 GOLOS")
+                             ("received_vesting", "0.0000 GOLOS")
+                             ("unlocked_limit", "0.0000 GOLOS")
+                             );
```

Comment:
previous ≈40 lines duplicated in lot of tests. please, create "prepare_balances" function in `golos_vesting_tester` and move them there